### DISCUSSION TITLE
General Chat

### DISCUSSION THREAD

**Discussion by steakhause** [inmybluesocks/inmybluesocks#7] [General]
"Your repository has been cloned" just means that I copied the code to my computer using git. [Here's a video](https://jason.handles.tech/videos/2023-02-23_09-45-09.mp4) of what I mean.

**Comment by inmybluesocks** [comment]
something is going on with my external command prompt. it won't let me do any commands. i don't understand why.
C:\Users\charl>../
'..' is not recognized as an internal or external command,
operable program or batch file.

**Reply by steakhause** [reply]
Assuming that you are trying to go up a level in the directory, you need "cd" before the "../". cd is Change Directory. 

cd ../

**Reply by inmybluesocks** [reply]
ok. ya. have to refresh myself on how to do command prompts
but even i do ls

VC:\Users\charl\Documents\github> rm -rf inmybluesocks
it comes up with a whole lot of red
it doesn't recognize -rf I guess? it's in gray. just those letters
great looking vid. hard to read what the command line says on pause though

**Reply by inmybluesocks** [reply]
wondering how you got to (main)
when I do that it just brings me into the inmybluesocks dir

**Reply by inmybluesocks** [reply]
what is that prediction extension in the command line? is it pylance

**Reply by inmybluesocks** [reply]
gitblame but still won't work

**Reply by steakhause** [reply]
I'm on linux. I don't believe the rm -rf thing works on Windows. You don't need that though. I had already cloned the repo and was deleting it to show the process from the beginning again. 

The parts you would do to download to a new computer and continue working on a particular branch are:

git clone git@github.com:inmybluesocks/inmybluesocks.git   (This copies the repo to your computer.)
cd inmybluesocks  (Go into the directory that you just created)
git branch -r  (shows all branches of the repository on github, usually referred to as "remote".)
git branch (tells you which branch you are currently working on.)
ls  (list the folders and files in the current directory)
git checkout in-the--gardens (change to the "in-the--gardens" branch)
git branch ( just verifies that you changed to the new branch.)
ls (list what's in the current branch)

Most of the time you will just use:
git pull origin ##WhateverBranchYouAreWorkingOn## (updates the files on your "local" computer with what is on github)
git commit -am "some message here" (Adds new files to the repo on your computer and gets them ready to go to gitHub)
git push origin ##WhateverBranchYouAreWorkingOn## (sends the changes to gitHub)

**Reply by steakhause** [reply]
I missed a couple of your comments. 

"main" is your default branch. When you clone the repo, you will automatically end up on that branch unless you specify another specifically.

**Reply by inmybluesocks** [reply]
when i try to get the branch it says permission denied

Warning: Permanently added 'github.com,140.82.114.4' (ECDSA) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

**Reply by inmybluesocks** [reply]
how do i get access rights?

**Reply by inmybluesocks** [reply]
Ok.  This is what I found/ recommended 
Acer 2023 Spin 2-in-1 714 Chromebook 14"
What do you think?

**Reply by inmybluesocks** [reply]
hooked up the rasp pi 4 but it still says that permission denied by github when i try to do the 1st line

**Reply by inmybluesocks** [reply]
am i getting this message b/c the rasp pi ssh is already directed to my laptop and cannot make anymore ssh requests? no idea

**Reply by inmybluesocks** [reply]
need a public key for repository in deploy keys. don't know how to make a new ssh key

**Reply by inmybluesocks** [reply]
do i have a github email on here?

**Reply by inmybluesocks** [reply]
ok. got the key but if i just put in
SHA256:VV6fkLDSToFCAI6YkVK5xICUODi5beC+D3y++0FdzBk pi@raspberrypi
into the inmybluesocks deploy but no-go

**Reply by inmybluesocks** [reply]
got the key. got the deploy set up
now it says
pi@raspberrypi:~/Documents/clone $ git clone git@github.com:inmybluesocks/inmybluesocks.git ~
fatal: destination path '/home/pi' already exists and is not an empty directory.
pi@raspberrypi:~/Documents/clone $

**Reply by steakhause** [reply]
https://jason.handles.tech/videos/2023-02-24_13-38-31.mp4

It sounds like you are past this part but just in case, give that video a look.

**Reply by steakhause** [reply]
The message that you are receiving "'/home/pi' already exists" means that it is trying to clone your repo into /home/pi/ instead of the folder that you are in, /Documents/clone/. That's strange. I'm not sure why that would happen. The line through the text makes it hard to read. Is there anything behind "git clone git@github.com:inmybluesocks/inmybluesocks.git"?

I'm not sure what terminal you are using but it seems weird that it shows you in /Documents/clone instead of ~/Documents/clone

Some things I would check. 

pwd  (This should return /home/pi/Documents/clone)
ls -al (The permissions listed should be drwxrwxr-x)
git config --global user.name (should be your git user)
git config --global user.email (should be your email)

**Reply by inmybluesocks** [reply]
I just went to the clone directory b/c I thought I needed to direct where to put the folder. I made that folder. If i do that github clone in either the vscode terminal or the rasp pi 4. Terminal. That error code shows up

**Reply by inmybluesocks** [reply]
My personal email or is there a github email?

**Reply by steakhause** [reply]
Your personal email.

**Comment by steakhause** [comment]
The prediction extension that I am using is called fish. The first two things I download when I install a new OS is terminator (a different terminal) and fish.

**Comment by inmybluesocks** [comment]


**Reply by inmybluesocks** [reply]
HOw much space should i allocate for my VM? how much free space should i keep as leftover for the main portion?

**Reply by steakhause** [reply]
This is what I always went to for creating a ssh key. Leave the password blank. https://www.digitalocean.com/community/tutorials/how-to-set-up-ssh-keys-2

You should be able to copy your key by typing  `cat ~/.ssh/id_rsa.pub` into the command prompt. That's what you want to paste in as your key in github.

For the VM, I usually allocate about 60G and allow it to scale dynamically. If you are going to be using the VM like a regular computer, browsing the web with a bunch of tabs open, text editor, command line open, etc. I usually split my RAM in half and allocate two processor cores.  If you are only going to be doing the coding, no web, you are fine with 2G of RAM.

**Reply by inmybluesocks** [reply]
yes
the key is SHA256...

git hub deploy key format says that it needs to be in   openssh public key format.  iv'e tried ssh-rsa SHA256.... & ssh-rsa.SHA256...
don't really know what the format looks like.

when i try to copies it says no identities found.
the key does not save in my ssh. folder. it puts it in my main pi directory.
thus when i try to copy the file
 ssh-copy-id pi@raspberrypi
it says no identities found

**Reply by inmybluesocks** [reply]
the old computer only has about 40gigs left after i took all the pis off and the pi sucks for coding
the new one i can divide
any good articles on how to set up a VM or the splitting of my RAM. don't know anything about splitting RAM. the new one has 20gigs of ram

**Reply by inmybluesocks** [reply]
no such file when coping using cat

**Comment by inmybluesocks** [comment]
still don't know what to do about the VMware. it looks like you have to pay now. i don't know

**Reply by inmybluesocks** [reply]
vnc viewer

**Comment by steakhause** [comment]
It appears that your default ssh key is not the default name of id_rsa.pub. You will want to remember that as it's likely to trip you up again at some point. 

Try this command for getting your key. This should be the key that is set as default for your user. 

`ssh-agent sh -c 'ssh-add; ssh-add -L'`

**Reply by steakhause** [reply]
Be sure not to copy too much of what is returned. You want everything after and including ssh-rsa. The top part will tell you the name and location of your key.

**Reply by steakhause** [reply]
Try ```
`ssh-agent sh -c 'ssh-add; ssh-add -L'`
```

The WSL thing that you're using is pretty new. I've never messed with it. I'm not sure what, if anything, is different.

**Reply by inmybluesocks** [reply]
this is what i'm getting with that. i'm using git bash terminal b/c i thought what they wanted me to download as a terminal

$ ssh-agent sh -c 'ssh-add; ssh-add -L'
The agent has no identities.

**Comment by inmybluesocks** [comment]
i'm working on trying to get my new computer to github using gitbash. i made the key fingerprint but when i try to cat the key it says
fjc22@charfru MINGW64 ~
$ cat ~/.ssh/id_one1.pub
cat: /c/Users/fjc22/.ssh/id_one1.pub: No such file or directory
what do you think?
no idea where i would go to find the ssh folder in windows to see if it's really there or not
